server/users/userController.js

```diff
@@ -17,8 +17,8 @@ module.exports.signin = function(req, res, next) {
         res.status(401).send('Bad request: User not found');
       } else {
         return user.comparePasswords(password)
-          .then(function(user) {
-            if (user) {
+          .then(function(validPassword) {
+            if (validPassword) {
```

Comment:
The returned variable is not a user here, but its a boolean for if the password was the same or not. I was getting back as the token 'true' or 'false' when decoded and wondered why. The user variable was masking the outer user variable.